### PR TITLE
Add draw_timeline binary

### PR DESCRIPTION
**Pull request by bojanserafimov**
Example output after running pgbench on with scale 10GB:
Screenshot from 2022-10-24 16-48-48

See the legend in the docstring for help on how to interpret the image.

This tool has been very helpful to me and I figured it's worth committing into the repo. I haven't worked much on code quality (not sure if it's worth the investment), but if it's unreadable let me know and I'll fix it.

### CODE REVIEW

**Comment by hlinnaka** [comment]
I love it! There's a lot of obvious next steps and improvements: display filenames as tooltips as you hover over a layer, parse and show relation IDs and block ranges, etc. etc. Let's get this in, so we can start improving it step by step.

File: pageserver/src/bin/draw_timeline_dir.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    ops::Range,
+};
+use svg_fmt::{rectangle, rgb, BeginSvg, EndSvg, Fill, Stroke};
+use utils::{lsn::Lsn, project_git_version};
+
+project_git_version!(GIT_VERSION);
+
+// Map values to their compressed coordinate - the index the value
+// would have in a sorted and deduplicated list of all values.
+fn build_coordingate_compression_map<T: Ord + Copy>(coords: Vec<T>) -> BTreeMap<T, usize> {
```
Comment (hlinnaka) [inline pageserver/src/bin/draw_timeline_dir.rs:39]:
```suggestion
fn build_coordinate_compression_map<T: Ord + Copy>(coords: Vec<T>) -> BTreeMap<T, usize> {
```

File: pageserver/src/bin/draw_timeline_dir.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
+//! A tool for visualizing the arrangement of layerfiles within a timeline.
+//!
+//! It reads filenames from stding and prints a svg on stdout. The image is a plot in
```
Comment (hlinnaka) [inline pageserver/src/bin/draw_timeline_dir.rs:3]:
```suggestion
//! It reads filenames from stdin and prints a svg on stdout. The image is a plot in
```

**Review by hlinnaka — APPROVED** [review]

**Comment by SomeoneToIgnore** [comment]
Might make sense to add it into the release deployment if we want to be able to run it anytime on staging/prod.
See `pageserver_binutils` usage for places where you want to add it for this purpose.

**Pull request merged**